src/openrct2/EditorObjectSelectionSession.cpp

```diff
@@ -375,6 +376,14 @@ void reset_selected_object_count_and_size()
     }
 }
 
+void finish_object_selection(const RCT2_EDITOR_STEP editorStep)
+{
+    set_every_ride_type_invented();
+    set_every_ride_entry_invented();
+    gS6Info.editor_step = editorStep;
+    gfx_invalidate_screen();
```

Comment:
You put the code from the Track Designer here, which now causes this to run in _all_ situations. You need to copy the whole if-else construct.